src/explore-education-statistics-common/src/modules/table-tool/components/TableToolWizard.tsx

```diff
@@ -0,0 +1,359 @@
+  slug: string;
+}
+
+interface TableToolState {
+  query: TableDataQuery | undefined;
+  createdTable: FullTable | undefined;
+  validInitialQuery: TableDataQuery | undefined;
+  dateRange: DateRangeState;
+  locations: Dictionary<LocationFilter[]>;
+  subjectId: string;
+  subjectMeta: PublicationSubjectMeta;
+}
+
+type PartialState = {
+  [P in keyof TableToolState]?: TableToolState[P];
+};
+
+export interface FinalStepProps {
+  publication?: Publication;
+  createdTable?: FullTable;
+  tableHeadersConfig?: TableHeadersConfig;
+  tableHeaders?: TableHeadersFormValues;
+  query?: TableDataQuery;
+}
+
+interface Props {
+  themeMeta: ThemeMeta[];
+  publicationId?: string;
+  releaseId?: string;
+
+  finalStep?: (props: FinalStepProps) => ReactElement;
+
+  initialQuery?: TableDataQuery;
+  onTableConfigurationChange?: (props: FinalStepProps) => void;
+}
+
+const TableToolWizard = (props: Props) => {
+  const {
+    themeMeta,
+    publicationId,
+    releaseId,
+    finalStep,
+    initialQuery,
+    onTableConfigurationChange,
+  } = props;
+
+  const [publication, setPublication] = React.useState<Publication>();
+  const [subjects, setSubjects] = React.useState<PublicationSubject[]>([]);
+
+  const [initialStep, setInitialStep] = React.useState(1);
+
+  const getInitialState = () => {
+    return {
+      initialStep: 1,
+      subjectId: '',
+      subjectMeta: getDefaultSubjectMeta(),
+      locations: {},
+      dateRange: {},
+      createdTable: undefined,
+      query: undefined,
+      validInitialQuery: undefined,
+    };
+  };
+
+  const [tableToolState, setTableToolState] = React.useState<TableToolState>(
+    () => {
+      return getInitialState();
+    },
+  );
+
+  React.useEffect(() => {
+    if (releaseId) {
+      tableBuilderService
+        .getReleaseMeta(releaseId)
+        .then(({ subjects: releaseSubjects }) => {
+          setSubjects(releaseSubjects);
+        });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [releaseId]);
+
+  React.useEffect(() => {
+    const currentlyLoadingQuery = {
+      releaseId,
+      initialQuery,
+    };
+    setInitialStep(1);
+    setTableToolState({
+      subjectMeta: getDefaultSubjectMeta(),
+      dateRange: {},
+      locations: {},
+      subjectId: '',
+      query: undefined,
+      createdTable: undefined,
+      validInitialQuery: undefined,
+    });
+
+    initialiseFromInitialQuery(releaseId, initialQuery).then(state => {
+      // make sure nothing changed in the component while we were processing the initialisation
+      if (
+        currentlyLoadingQuery.releaseId === releaseId &&
+        currentlyLoadingQuery.initialQuery === initialQuery
+      ) {
+        const { initialStep } = state;
+
+        setInitialStep(initialStep);
+
+        setTableToolState(state);
+
+        if (onTableConfigurationChange)
+          onTableConfigurationChange({ ...state });
+      }
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Ditto

Reply:
onTableConfigurationChange

Reply:
It doesn't matter if it changes. There is a guard against it

Reply:
ACtually... no... updated to use a callback and added as a dependency